main/main.cpp

```diff
@@ -3308,6 +3308,11 @@ int Main::start() {
 		{
 			Ref<DirAccess> da = DirAccess::open(doc_tool_path);
 			ERR_FAIL_COND_V_MSG(da.is_null(), EXIT_FAILURE, "Argument supplied to --doctool must be a valid directory path.");
+			// Ensure that doctool is running in the root dir, but only if
+			// user did not manually specify a path as argument.
+			if (doc_tool_path == ".") {
```

Comment:
Won't it break if using `godot --doctool .` when you actually want to generate doc templates in your cwd?

Reply:
Yeah though you could in theory use `./`

I could change it so `--doctool` sets the path to empty string, and have a flag like `bool doc_tool_enabled` to detect if doctool should run (instead of string is empty check)?

Reply:
Instead of relying on `doc_tool_path == "."` I switched to a boolean that's set whenever the cwd is *implicitly* used.